include/ebpf_api.h

```diff
@@ -417,6 +417,20 @@ extern "C"
         size_t attach_parameters_size,
         _Outptr_ struct bpf_link** link) EBPF_NO_EXCEPT;
 
+    /**
+     * @brief Attach an eBPF program by program file descriptor and return
+     * the link as a file descriptor.
+     *
+     * @see ebpf_program_attach_by_fd
+     */
+    _Must_inspect_result_ ebpf_result_t
+    ebpf_program_attach_as_fd(
```

Comment:
Not a great name, `ebpf_program_attach_as_fd` which already exists would make more sense imo. Open to bikeshedding.

Reply:
maybe `ebpf_program_attach_by_fd_and_get_fd`?
It's wordy but the "by fd" in 413 refers to the first input, which still applies in this prototype

Reply:
Actually, I'm currently working on `ebpf_object_load_native_fds`.

Reply:
To me, "attach fds" and "load ... fds" sound like they attach or load multiple fds, which is not the case.

Reply:
Can we break API and rename `_by_fd` to `_from_fd` or so?

Reply:
Or `ebpf_program_attach_by_fds`?

Reply:
Changing an existing ebpf_ name can be done without a breaking change by also defining a deprecated alias, as long as the API signature is the same.

ebpfapi\Source.def shows we have already done this three times:
```
 bpf_program__get_type=bpf_program__type
 ebpf_enumerate_sections = ebpf_enumerate_programs
 ebpf_free_sections = ebpf_free_programs
```
And in include\ebpf_api.h you'll see the latter two `@deprecated` functions (and the first one in include\bpf\libbpf.h)

Reply:
This doesn't let use reuse the original name however?

Reply:
@dthaler can you take another look?